.github/workflows/kuksa_databroker-cli_build.yml

```diff
@@ -80,19 +85,56 @@ jobs:
           ~/.cargo/registry/cache/
           ~/.cargo/git/db/
           target/
-        key: databroker-cli-release-${{ runner.os }}-${{ hashFiles('**/Cargo.lock') }}
-
+        key: databroker-release-${{ matrix.platform.name }}-${{ hashFiles('**/Cargo.lock') }}
     - name: Install build prerequisites
       working-directory: ${{github.workspace}}/
       run: |
-        which cargo-license || cargo install cargo-license
         which cross || cargo install cross
-    - name: Building
+        cargo install cargo-license cargo-cyclonedx
+        pip install "git+https://github.com/eclipse-kuksa/kuksa-common.git@6f3d7627760582d8ba83cc8a0f7449d00fffee84#subdirectory=sbom-tools"
+    - name: Build
       working-directory: ${{github.workspace}}/
+      env:
+        KUKSA_DATABROKERCLI_SBOM: y
+      run: |
+        ./build-databroker-cli.sh ${{ matrix.platform.name }}
```

Comment:
I think it would be a lot better if we avoid reintroducing SBOM generation / license checking to the build step. All the other "sanity checks" are run in parallel, i.e. unit tests / formatting etc. The artifacts (SBOM) are not needed until after the build has completed anyway (for the release artifact creation).

Reply:
Yes, but it would be a bit harder to refactor and I guess not save as much time overall, as you still need to do it 3 times (the sbom is different, depending on the target, i.e. if you compare the sbom.json in the artifacts you will see different dependencies)

Reply:
Yes, but that uses the old bom script, that does not take into account that SBOM is different, depending on the target architecture (i.e. it is only run once). The goal is to have an SBOM _exactly_ for a delivered binary.

No _theoretically_ it might be split (i.e. running this in parallel also for the 3 currently build architectures), but I think the benefits far outweigh the risks of mis-using this. Because you would need to "combine" the correct SBOMs with the correct builds. If you would use a different set of "feature" flags in cargo or potentially another rust version on another runner, this will not match. So in principle SBOMs should be generated "as close to the build" as possible. Currently by running them in the same script, they run exactly in the same environment as the build. And that is also true, when using this outside CI.

Reply:
It would be good if generating the SBOM was at least a separate step.. Currently, I don't know whether it's that or something with the compilation that is causing the prolonged build time.

This PR:
![Build time - PR](https://github.com/eclipse-kuksa/kuksa-databroker/assets/1281079/96ab7d85-f642-4136-a77d-9e959a74fd96)


Baseline: 

![Build time - Baseline](https://github.com/eclipse-kuksa/kuksa-databroker/assets/1281079/2ba4896f-68a1-46cb-a800-7fb76a6c0304)